sdk/cli/src/commands/hasura/track-table.ts

```diff
@@ -0,0 +1,158 @@
+import { createExamples } from "@/utils/commands/create-examples";
+import { getApplicationOrPersonalAccessToken } from "@/utils/get-app-or-personal-token";
+import { getHasuraEndpoints } from "@/utils/get-cluster-service-endpoint";
+import { Command } from "@commander-js/extra-typings";
+import { createSettleMintClient } from "@settlemint/sdk-js";
+import { loadEnv } from "@settlemint/sdk-utils/environment";
+import { intro, note, outro, spinner } from "@settlemint/sdk-utils/terminal";
+import type { DotEnv } from "@settlemint/sdk-utils/validation";
+
+export function hasuraTrackTableCommand() {
+  return new Command("track-table")
+    .alias("tt")
+    .description("Track a table in Hasura")
+    .usage(
+      createExamples([
+        {
+          description: "Track a table in the public schema",
+          command: "hasura track-table --schema public --table user",
+        },
+
+        {
+          description: "Track a table and accept default values",
+          command: "hasura track-table --schema public --table payments --accept-defaults",
+        },
+      ]),
+    )
+    .requiredOption("--schema <schema>", "Schema name of the table (e.g., public)")
+    .requiredOption("--table <table>", "Name of the table to track")
+    .option("-a, --accept-defaults", "Accept the default and previously set values")
+    .action(async ({ schema, table, acceptDefaults }) => {
+      intro("Tracking table in Hasura");
+
+      const env: Partial<DotEnv> = await loadEnv(false, false);
+      const applicationUniqueName = env.SETTLEMINT_APPLICATION;
+      if (!applicationUniqueName) {
+        return missingApplication();
+      }
+
+      const selectedInstance = await instancePrompt(env, true);
+      const accessToken = await getApplicationOrPersonalAccessToken({
+        env,
+        instance: selectedInstance,
+        prefer: "application",
+      });
+
+      const settlemint = createSettleMintClient({
+        accessToken,
+        instance: selectedInstance,
+      });
+
+      const integrationTools = await serviceSpinner("integration tool", () =>
+        settlemint.integrationTool.list(applicationUniqueName),
+      );
+
+      const hasura = await hasuraPrompt({
+        env,
+        integrations: integrationTools,
+        accept: acceptDefaults,
+        isRequired: true,
+      });
+
+      if (!hasura) {
+        return nothingSelectedError("Hasura instance");
+      }
+
+      const hasuraEndpoints = getHasuraEndpoints(hasura);
+      const hasuraGraphqlEndpoint = hasuraEndpoints.SETTLEMINT_HASURA_ENDPOINT;
+      const hasuraAdminSecret = hasuraEndpoints.SETTLEMINT_HASURA_ADMIN_SECRET;
+
+      if (!hasuraGraphqlEndpoint || !hasuraAdminSecret) {
+        return note("Could not retrieve Hasura endpoint or admin secret. Please check your configuration.");
+      }
+
+      // Convert GraphQL endpoint to Query endpoint
+      const baseUrl = new URL(hasuraGraphqlEndpoint);
```

Comment:
once https://github.com/settlemint/btp/pull/6431 is in, i can use it to construct the query endpoint